RELEASE_PROCESS.md

```diff
@@ -0,0 +1,268 @@
+# Podman Releases
+
+## Overview
+
+Podman (and podman-remote) versioning is mostly based on semantic-versioning
```

Comment:
Should we link to what semantic versioning is? 
```suggestion
Podman (and podman-remote) versioning is mostly based on [semantic-versioning](https://semver.org/)
```

Reply:
Good idea